### PR TITLE
change export style and export StringUtility

### PR DESCRIPTION
**Pull request by codeBelt**

### CODE REVIEW

**Comment by codeBelt** [comment]
https://hackernoon.com/import-export-default-require-commandjs-javascript-nodejs-es6-vs-cheatsheet-different-tutorial-example-5a321738b50f

**Comment by JimSchofield** [comment]
Looks good.  I don't know if there's a way to export default and export a module separately

**Comment by codeBelt** [comment]
@JimSchofield there is but I don't know you would import it with require. Below is the only way I know how to do it.

```
const {generateTemplateFiles, StringUtility} = require('generate-template-files');
```

You could do it this way but I want it to work with `require`
```
import generateTemplateFiles, {StringUtility} from 'generate-template-files';
```
@ccheney

File: examples/tools/generate.js [hunk 1]
```diff
@@ -131,3 +124,24 @@ generateTemplateFiles([
         },
     },
 ]);
+
+/*
+ * NOTE: there is many ways you can do this. This is just an example on how you might approch it.
```
Comment (codeBelt) [inline examples/tools/generate.js:130]: @JimSchofield I added an example here

**Comment by codeBelt** [comment]
@JimSchofield 

I was thinking of changing the filesAndFolders to only list the files created. Here is what it outputs currently.
```
{
  output: {
    path: './src/stores/some-thing',
    filesAndFolders: [
      'SomeThingModule.ts',
      'SomeThingModuleAction.ts',
      'SomeThingModuleGetter.ts',
      'SomeThingModuleMutation.ts',
      'SomeThingService.ts',
      'models',
      'models/actions',
      'models/actions/ISomeThingState.ts',
      'models/actions/OtherThingResponseModel.ts',
      'models/getters'
    ]
  },
  stringReplacers: [
    {
      slot: '__store__',
      slotValue: 'some thing'
    },
    {
      slot: '__model__',
      slotValue: 'other thing'
    }
  ]
}
```

Here is the new way:
```
{
  output: {
    path: './src/stores/some-thing',
    filesAndFolders: [
      './src/stores/some-thing/SomeThingModule.ts',
      './src/stores/some-thing/SomeThingModuleAction.ts',
      './src/stores/some-thing/SomeThingModuleGetter.ts',
      './src/stores/some-thing/SomeThingModuleMutation.ts',
      './src/stores/some-thing/SomeThingService.ts',
      './src/stores/some-thing/models/actions/ISomeThingState.ts',
      './src/stores/some-thing/models/actions/OtherThingResponseModel.ts',
    ]
  },
  stringReplacers: [
    {
      slot: '__store__',
      slotValue: 'some thing'
    },
    {
      slot: '__model__',
      slotValue: 'other thing'
    }
  ]
}
```

Part of me thinks I should just leave it so it has everything that was outputted and the developer can choose what to do.

**Comment by JimSchofield** [comment]
I think explicitly listing out files copied and not including folders makes more sense.  You can figure out the folders based on the files copied.

For string replacers, I would suggest `slotDefaultValue` because that would call out that those values are determined by the defaultCase config item.

I like the description on the actual data structure returned.

**Pull request merged**